lib/puppet/util/windows/user.rb

```diff
@@ -145,6 +145,90 @@ def load_profile(user, password)
   end
   module_function :load_profile
 
+  def get_rights(name)
+    rights = []
+    rights_pointer = FFI::MemoryPointer.new(:pointer)
+    number_of_rights = FFI::MemoryPointer.new(:ulong)
+
+    policy_handle = open_new_lsa_policy
+    sid_pointer = Puppet::Util::Windows::SID::Principal::lookup_sid_pointer(name)
+
+    result = LsaEnumerateAccountRights(policy_handle.read_pointer, sid_pointer, rights_pointer, number_of_rights)
+
+    win32_error = LsaNtStatusToWinError(result)
+    return rights if win32_error == 2 # No rights found error
+
+    check_lsa_nt_status_and_raise_failures(result, "`LsaEnumerateAccountRights` method failed")
+
+    # https://stackoverflow.com/questions/24461466/how-to-use-ruby-ffi-to-read-array-of-static-structures
+    number_of_rights.read_ulong.times do |i|
+      right = LSA_UNICODE_STRING.new(rights_pointer.read_pointer + i * LSA_UNICODE_STRING.size)
+      rights << right[:Buffer].read_arbitrary_wide_string_up_to
+    end
+
+    result = LsaFreeMemory(rights_pointer.read_pointer)
+    check_lsa_nt_status_and_raise_failures(result, "`LsaFreeMemory` method failed")
+
+    LsaClose(policy_handle.read_pointer)
+
+    rights
+  end
+  module_function :get_rights
+
+  def set_right(name, right)
+    right_pointer = new_right_pointer(right)
+    policy_handle = open_new_lsa_policy
+    sid_pointer = Puppet::Util::Windows::SID::Principal::lookup_sid_pointer(name)
+
+    result = LsaAddAccountRights(policy_handle.read_pointer, sid_pointer, right_pointer, 1)
+    check_lsa_nt_status_and_raise_failures(result, "`LsaAddAccountRights` method failed")
+
+    LsaClose(policy_handle.read_pointer)
+  end
+  module_function :set_right
+
+  def remove_right(name, right)
```

Comment:
Works but not used yet. Need to figure out the extra parameter usage/implementation (for the `inclusive` and `minimum` cases).